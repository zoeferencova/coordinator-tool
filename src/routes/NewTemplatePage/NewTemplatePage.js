import React, { useState, useContext } from 'react';
import { useNavigate } from 'react-router-dom';
import { Link } from 'react-router-dom'
import AppContext from '../../contexts/contexts'
import { ButtonDark, ButtonLight, Input, Textarea, Error } from '../../components/Utils/Utils'

import styles from './NewTemplatePage.module.css';
import TemplateService from '../../services/template-service';

const NewTemplatePage = ({ onboarding }) => {
    const context = useContext(AppContext);
    const [error, setError] = useState(null);
    const navigate = useNavigate();

    const handlePostTemplate = async e => {
        e.preventDefault();
        const template_name = e.target.template_name.value;
        const template_subject = e.target.template_subject.value;
        const template_content = e.target.template_content.value;

        const template = { template_name, template_subject, template_content }

        TemplateService.addTemplate(template)
            .then(res => !res.ok ? res.json().then(e => Promise.reject(e)) : res.json())
            .then(newTemplate => handlePostSuccess(newTemplate))
            .catch(res => {
                if (res.error.message === "Missing 'template_name' in request body") {
                    setError("Missing template name")
                } else if (res.error.message === "Missing 'template_subject' in request body") {
                    setError("Missing template subject")
                } else if (res.error.message === "Missing 'template_content' in request body") {
                    setError("Missing template body")
                } else {
                    setError(res.error.message)
                }
            })
    }

    const handlePostSuccess = template => {
        context.addTemplate(template)
        navigate('/templates')
    }

    return (
        <div className={`container ${onboarding ? styles.onboardingOpen : ""}`}>
            <main className="content">
                <form className="form" onSubmit={e => handlePostTemplate(e)}>
                    <h2>New Template</h2>
                    {error && <Error style={{ marginRight: "25px", marginLeft: "25px" }} error={error} />}

                    <div className="form-item">
                        <label htmlFor="template_name">Name</label>
                        <Input type="text" name='template_name' id='template_name'></Input>
                    </div>
                    <div className="form-item">
                        <label htmlFor="template_subject">Subject</label>
                        <Input type="text" name='template_subject' id='template_subject'></Input>
                    </div>
                    <br></br>
                    <div className="form-item">
                        <label htmlFor="template_content">Body</label>
                        <Textarea name="template_content" id="template_content"></Textarea>
                    </div>
                    <p className={styles.instructions}>Note: For template functionality, use [CONTACT] for contact name, [PROJECT] for project name, and [PM] for PM name.</p>
                    <div className="form-buttons">
                        <Link to='/templates'><ButtonLight>Cancel</ButtonLight></Link>
                        <ButtonDark type='submit'>Create Template</ButtonDark>
                    </div>
                </form>

            </main>
        </div>
    )
}

export default NewTemplatePage;