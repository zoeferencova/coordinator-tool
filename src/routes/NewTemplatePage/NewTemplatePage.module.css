.input {
    margin-top: 10px;
    width: 300px;
}

.textAreaSection {
    display: flex;
    justify-content: center;
    margin-bottom: 20px;
}

.textArea {
    height: 500px;
}

.textAreaLabel {
    margin-right: 5px;
}

.instructions {
    font-size: 12px;
    color: gray;
    padding-top: 0px;
    margin-left: 25px;
}