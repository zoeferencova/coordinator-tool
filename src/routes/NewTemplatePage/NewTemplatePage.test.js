import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NewTemplatePage from './NewTemplatePage'

describe(`NewTemplatePage component`, () => {
  it('renders a .NewTemplatePage by default', () => {
    const wrapper = shallow(<NewTemplatePage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})