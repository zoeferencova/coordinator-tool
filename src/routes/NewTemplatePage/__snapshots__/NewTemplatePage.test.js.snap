// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTemplatePage component renders a .NewTemplatePage by default 1`] = `
<div
  className="container"
>
  <main
    className="content"
  >
    <Header
      title="New Template"
    />
    <form
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="template_name"
        >
          Name: 
        </label>
        <Input
          className="input"
          id="template_name"
          name="template_name"
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="template_subject"
        >
          Subject: 
        </label>
        <Input
          className="input"
          id="template_subject"
          name="template_subject"
          type="text"
        />
      </div>
      <br />
      <div
        className="textAreaSection"
      >
        <label
          className="textAreaLabel"
          htmlFor="template_content"
        >
          Body: 
        </label>
        <Textarea
          className="textArea"
          id="template_content"
          name="template_content"
        />
      </div>
      <p
        className="instructions"
      >
        Note: For template functionality, use [ADVISOR] for advisor name, [PROJECT] for project name, and [PM] for PM name.
      </p>
      <div>
        <Link
          to="/email"
        >
          <Button>
            Cancel
          </Button>
        </Link>
        <Button
          type="submit"
        >
          Create Template
        </Button>
      </div>
    </form>
  </main>
  <NavBar
    className="nav"
  />
</div>
`;
